pkg/media/ivfreader/ivfreader.go

```diff
@@ -0,0 +1,133 @@
+package ivfreader
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+const (
+	ivfFileHeaderSignature = "DKIF"
+	ivfFileHeaderSize      = 32
+	ivfFrameHeaderSize     = 12
+)
+
+// IVFFileHeader 32-byte header for IVF files
+// https://wiki.multimedia.cx/index.php/IVF
+type IVFFileHeader struct {
+	signature     string // 0-3
+	version       uint16 // 4-5
+	headerSize    uint16 // 6-7
+	fourcc        string // 8-11
+	width         uint16 // 12-13
+	height        uint16 // 14-15
+	timebaseDenum uint32 // 16-19
+	timebaseNum   uint32 // 20-23
+	numFrames     uint32 // 24-27
+	unused        uint32 // 28-31
+}
+
+// IVFFrameHeader 12-byte header for IVF frames
+// https://wiki.multimedia.cx/index.php/IVF
+type IVFFrameHeader struct {
+	frameSize uint32 // 0-3
+	timestamp uint64 // 4-11
+}
+
+// IVFReader is used to read IVF files and return frame payloads
+type IVFReader struct {
+	stream io.Reader
+}
+
+// NewWith returns a new IVF reader and IVF file header
+// with an io.Reader input
+func NewWith(in io.Reader) (*IVFReader, *IVFFileHeader, error) {
+	if in == nil {
+		return nil, nil, fmt.Errorf("stream is nil")
+	}
+
+	reader := &IVFReader{
+		stream: in,
+	}
+
+	// TODO: Return file header somewhere else?
```

Comment:
Maybe make the header a member of `IVFReader` maybe? Can make it non-public, and be a method.

Personally I think it is fine to return it. Users can just do `_` and ignore it if they want!

Reply:
Sounds good! I’ll keep it as-is then.